api/data_source.go

```diff
@@ -40,6 +41,32 @@ const (
 	RO DataSourceType = "RO"
 )
 
+// DataSourceOptions is the options for a data source.
+type DataSourceOptions struct {
```

Comment:
Had a second thoughts about this after reading the doc https://www.mongodb.com/developer/products/mongodb/srv-connection-strings/

My current understanding is it's unlikely that user would choose mongodb:// at the instance level, while choosing mongodb+srv:// at the data source level (or vice versa). Thus having this option at the data source level is kind of duplicated?

The further question is would the underlying client library handles it transparently. So we just pass it a mongo.example.com, could it resolve?